pkgs/development/interpreters/php/default.nix

```diff
@@ -49,9 +52,9 @@ let
             "LDAP_DIR=${openldap.dev}"
             "LDAP_INCDIR=${openldap.dev}/include"
             "LDAP_LIBDIR=${openldap.out}/lib"
-            "--with-ldap-sasl=${cyrus_sasl.dev}"
+            (lib.optional stdenv.isLinux "--with-ldap-sasl=${cyrus_sasl.dev}")
             ];
-          buildInputs = [openldap cyrus_sasl openssl];
+          buildInputs = [openldap (lib.optional stdenv.isLinux cyrus_sasl) openssl];
```

Comment:
I think this is usually appended at the end:

```
++ lib.optional stdenv.isLinux cyrus_sasl
```

Otherwise it's just a list in a list.